cromwell_tools/cromwell_tools.py

```diff
@@ -327,3 +332,58 @@ def generate_auth_header_from_key_file(json_credentials):
     scopes = ['https://www.googleapis.com/auth/userinfo.profile', 'https://www.googleapis.com/auth/userinfo.email']
     credentials = ServiceAccountCredentials.from_json_keyfile_name(json_credentials, scopes=scopes)
     return {"Authorization": "bearer " + credentials.get_access_token().access_token}
+
+
+def _localize_file(url, target_directory='.'):
+    """Localize file url to a directory. Supports local files and http endpoints
+
+    :param str url: url of local or http target
+    :param str target_directory: directory to localize file to
+    :return:
+    """
+
+    if not os.path.isdir(target_directory):
+        raise NotADirectoryError(
+            'target_directory must be a valid directory on the local filesystem')
+
+    basename = os.path.basename(url)
+    target_file = os.path.join(target_directory, basename)
+
+    if url.startswith('http'):
+        data = download_http(url)
+        with open(target_file, 'wb') as f:
+            f.write(data)
+    else:
+        if not os.path.isfile(url):
+            raise FileNotFoundError(
+                'non-http files must point to a valid file on the local filesystem. Not found: %s'
+                % url)
+        else:
+            shutil.copy(url, target_file)
+
+
+def validate_workflow(wdl, womtool_path, dependencies_json=None):
+    """Validate a wdl workflow using cromwell womtool
+
+    :param str wdl: link or filepath to wdl file
+    :param str womtool_path: path to womtool.jar
+    :param str dependencies_json: file path to json file containing dependencies
+    :return:
+    """
+    temporary_directory = tempfile.mkdtemp()
+
+    _localize_file(wdl, temporary_directory)
+    wdl_basename = os.path.basename(wdl)
+
+    if dependencies_json is not None:
+        with open(dependencies_json, 'r') as f:
+            depenencies_map = json.load(f)
+        for url in depenencies_map.values():
+            _localize_file(url, temporary_directory)
+
+    os.chdir(temporary_directory)
+    p = Popen(['java', '-jar', os.path.expanduser(womtool_path), 'validate', wdl_basename],
```

Comment:
Could be even better to check if `java` is in the current system path.

Reply:
Made issue #27 , depends on #26 .